tests/test_plugin.py

```diff
@@ -0,0 +1,211 @@
+
+    @patch.object(ReportPortalPlugin, 'addSkip')
+    def test_addError_with_SkipTest_exception(self, mocked_addSkip):
+        err = (SkipTest, SkipTest('test addError with SkipTest exception'), Mock())
+
+        self.plugin.addError(self.test_object, err)
+
+        mocked_addSkip.assert_called_once_with(self.test_object)
+
+    @patch.object(ReportPortalPlugin, 'addDeprecated')
+    def test_addError_with_DeprecatedTest_exception(self, mocked_addDeprecated):
+        err = (DeprecatedTest, DeprecatedTest('test addError with DeprecatedTest exception'), Mock())
+
+        self.plugin.addError(self.test_object, err)
+
+        mocked_addDeprecated.assert_called_once_with(self.test_object)
+
+    def test__addError(self):
+        try:
+            raise TestException('')
+        except TestException:
+            err = sys.exc_info()
+        expected_test_err_value = err[1]
+        expected_test_err_info = str(err[0].__name__) + ":\n" + "".join(traceback.format_tb(err[2]))
+
+        self.plugin._addError(self.test_object, err)
+
+        self.assertEqual(expected_test_err_value, self.test_object.errors[0])
+        self.assertEqual(expected_test_err_info, self.test_object.errors[1])
+
+    @patch.object(ReportPortalPlugin, '_addError')
+    def test_addFailure(self, mocked__addError):
+        err = (TestException, TestException('test addError with an uncaught exception'), Mock())
+
+        self.plugin.addFailure(self.test_object, err)
+
+        self.assertEqual('failed', self.test_object.status)
+        mocked__addError.assert_called_once_with(self.test_object, err)
+
+    def test_end(self):
+        self.plugin.stdout.append(MagicMock())
+
+        self.plugin.end()
+
+        self.assertEqual(0, len(self.plugin.stdout))
+
+    def test___restore_stdout(self):
+        for _ in range(random.randint(1, 10)):
+            self.plugin.stdout.append(MagicMock())
+
+        self.plugin._restore_stdout()
+
+        self.assertEqual(0, len(self.plugin.stdout))
+
+    @patch.object(ReportPortalPlugin, '_restore_stdout')
+    def test_finalize(self, mocked__restore_stdout):
+        self.plugin.finalize(result=Mock())
+
+        self.plugin.service.finish_launch.assert_called_once_with()
+        self.plugin.service.terminate_service.assert_called_once_with()
+        mocked__restore_stdout.assert_called_once_with()
+
+    @patch.object(ReportPortalPlugin, 'setupLoghandler')
+    @patch.object(ReportPortalPlugin, 'start')
+    def test_start_test(self, mocked_start, mocked_setupLoghandler):
+        self.test_object.status = Mock()
+        self.test_object.errors = Mock()
+
+        self.plugin.startTest(self.test_object)
+
+        self.assertIsNone(self.test_object.status)
+        self.assertIsNone(self.test_object.errors)
+        self.plugin.service.start_nose_item.assert_called_once_with(self.plugin, self.test_object)
+        mocked_start.assert_called_once_with()
+        mocked_setupLoghandler.assert_called_once_with()
```

Comment:
I'd suggest using soft asserts here ( See delayed_asserts) to make sure we run through all of the checks even if any of them fails.